packages/react/src/components/Debug/FiberNavigator.ts

```diff
@@ -324,13 +324,33 @@ class FiberNavigator {
   }
 
   get instance() {
-    return this.isClassComponent
-      ? this.__fiber.stateNode
-      : this.isFunctionComponent // TODO: assumes functional component w/useRef
-      ? this.__fiber.memoizedState &&
-        this.__fiber.memoizedState.memoizedState &&
-        this.__fiber.memoizedState.memoizedState.current
-      : null
+    if (this.isClassComponent) {
+      return this.__fiber.stateNode
+    }
+
+    if (this.isFunctionComponent) {
+      // assumes functional component w/useRef
+      return this.findMemoizedState(this.__fiber.memoizedState)
+    }
+
+    return null
+  }
+
+  findMemoizedState(thing) {
```

Comment:
`node` may be?

```
Hooks state is represented by a recursive structure where:
- `memoizedState` is a current value if applicable
- `next` is next hook in order
```

Let's also add a descriptive comment

Reply:
thing is awesome :D